src/odemis/acq/stitching/_registrar.py

```diff
@@ -511,50 +464,34 @@ def _compute_registration(self, tile, row, col):
         # store the tile
         self.tiles[row][col] = tile
 
-        # expected tile position, if there would be no shift
-        # we avoid using tsize to not loose precision due to rounding
-        ox = int(col * (self.size[1] * (1 - self.ovrlp)))
-        oy = int(row * (self.size[0] * (1 - self.ovrlp)))
-
-        pos = None
-        pos2 = None
-        match = 0
-        match2 = 0
-
-        if self.pos_prev_x < col:
-            if col != 0 and self.tiles[row][col - 1] is not None:
-                # calculate the shift compared to the neighbor on the left if
-                # possible
-                pos, match = self._pos_to_left_right(row, col, LEFT)
-        elif self.pos_prev_x > col:
-            if col != self.nx and self.tiles[row][col + 1] is not None:
-                # calculate the shift compared to the neighbor on the right if
-                # possible
-                pos, match = self._pos_to_left_right(row, col, RIGHT)
+        if self.pos_prev_x < col and col != 0 and self.tiles[row][col - 1] is not None:
+            pos_hor, match_hor = self._register_horizontally(row, col, LEFT_TO_RIGHT)
+        elif self.pos_prev_x > col and col != self.nx and self.tiles[row][col + 1] is not None:
+            pos_hor, match_hor = self._register_horizontally(row, col, RIGHT_TO_LEFT)
+        else:
+            pos_hor, match_hor = None, 0
 
         if row != 0 and self.tiles[row - 1][col] is not None:
-            # calculate the shift compared to the neighbor on the top if
-            # possible
-            pos2, match2 = self._pos_to_top(row, col)
-
-        # get rid of extreme values
-        if (pos is None) and (pos2 is None):
-            pos = (ox, oy)
-        elif (pos is None):
-            pos = pos2
-        elif (pos2 is None):
-            pass
-        elif match > GOOD_MATCH and match2 > GOOD_MATCH:
-            # if the match is not really bad, then take the average position
-            pos = (int((pos[0] + pos2[0]) / 2), int((pos[1] + pos2[1]) / 2))
-        elif match2 > match:
-            # otherwise just keep the best one
-            pos = pos2
-        elif match2 == 0 and match == 0:
-            # no good value, fall back to given position
-            pos = (ox, oy)
+            pos_ver, match_ver = self._register_vertically(row, col)
+        else:
+            pos_ver, match_ver = None, 0
+
+        # Fallback to expected position if match is 0 (shift is larger than overlap size)
+        if ((pos_hor is None) and (pos_ver is None)) or (match_hor == 0 and match_ver == 0):
+            # expected tile position, if there would be no shift
+            ox = int(col * (self.size[1] * (1 - self.ovrlp)))
```

Comment:
Isn't it the same as `metadata[model.MD_POS]`? If it's not the same, why is that better than the metadata?